pennylane/devices/tests/test_templates.py

```diff
@@ -0,0 +1,995 @@
+            return qml.probs(u_tape.wires + v_wires)
+
+        def cost_lhst(parameters, v_function, v_wires, u_tape):
+            # pylint:disable=unsubscriptable-object
+            return (
+                1
+                - local_hilbert_test(
+                    v_params=parameters, v_function=v_function, v_wires=v_wires, u_tape=u_tape
+                )[0]
+            )
+
+        res = cost_lhst(
+            [3 * np.pi / 2, 3 * np.pi / 2, np.pi / 2],
+            v_function=v_function,
+            v_wires=[2, 3],
+            u_tape=u_tape,
+        )
+        expected = 0.5
+        assert np.isclose(res, expected, atol=tol(dev.shots))
+
+    def test_MERA(self, device, tol):
+        """Test the MERA template."""
+
+        def block(weights, wires):
+            qml.CNOT(wires=[wires[0], wires[1]])
+            qml.RY(weights[0], wires=wires[0])
+            qml.RY(weights[1], wires=wires[1])
+
+        n_wires = 4
+        n_block_wires = 2
+        n_params_block = 2
+        n_blocks = qml.MERA.get_n_blocks(range(n_wires), n_block_wires)
+        template_weights = [[0.1, -0.3]] * n_blocks
+        dev = device(n_wires)
+
+        dev = qml.device("default.qubit", wires=range(n_wires))
+
+        @qml.qnode(dev)
+        def circuit(template_weights):
+            qml.MERA(range(n_wires), n_block_wires, block, n_params_block, template_weights)
+            return qml.expval(qml.Z(1))
+
+        res = circuit(template_weights)
+        expected = 0.799260896638786
+        assert np.isclose(res, expected, atol=tol(dev.shots))
+
+    def test_MPS(self, device, tol):
+        """Test the MPS template."""
+
+        def block(weights, wires):
+            qml.CNOT(wires=[wires[0], wires[1]])
+            qml.RY(weights[0], wires=wires[0])
+            qml.RY(weights[1], wires=wires[1])
+
+        n_wires = 4
+        n_block_wires = 2
+        n_params_block = 2
+        n_blocks = qml.MPS.get_n_blocks(range(n_wires), n_block_wires)
+        template_weights = [[0.1, -0.3]] * n_blocks
+        dev = device(n_wires)
+
+        dev = qml.device("default.qubit", wires=range(n_wires))
+
+        @qml.qnode(dev)
+        def circuit(template_weights):
+            qml.MPS(range(n_wires), n_block_wires, block, n_params_block, template_weights)
+            return qml.expval(qml.Z(n_wires - 1))
+
+        res = circuit(template_weights)
+        expected = 0.8719048589118708
+        assert np.isclose(res, expected, atol=tol(dev.shots))
+
+    def test_MottonenStatePreparation(self, device, tol):
+        """Test the MottonenStatePreparation template."""
+        dev = device(3)
+        if dev.shots:
+            pytest.skip("test only works with analytic-mode simulations")
+
+        @qml.qnode(dev)
+        def circuit(state):
+            qml.MottonenStatePreparation(state_vector=state, wires=range(3))
+            return qml.state()
```

Comment:
Could we return `qml.probs()` instead?

Reply:
I opted for `state()` because I wanted to be sure that it was the actual inputted state, not just correct up to a phase. that was my motivation, but I can change it if you prefer

Reply:
My reason for preferring `probs` is then we can actually test it on hardware and with finite shots.  `qml.state()` works with a pretty limited subset of devices.

Reply:
i'll add a second test!